core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2379,8 +2382,13 @@ class SparkContext(config: SparkConf) extends Logging {
                 " parameter from breaking Spark's ability to find a valid constructor.")
           }
         }
-        listenerBus.addListener(listener)
-        logInfo(s"Registered listener $className")
+        logInfo(s"listener $className created")
+        listener
+      }
+      if (extraListeners.nonEmpty) {
+        val group = new FixGroupOfListener(extraListeners, "extraListeners")
```

Comment:
`FixGroupOfListener` is a bad class name. I'm not even sure what it's supposed to be, but the closest I can think of is `ListenerGroup`.

But perhaps this shouldn't be exposed at all. If you add a queue name to the listener registration method, you can hide this from callers altogether. That is, if I understood what this class is in the first place.

Then you wouldn't need `addIsolatedListener` either.